### PR TITLE
Simpler Swaps

### PR DESCRIPTION
**Pull request by mvader**
Now all of them are consistent and simpler

### CODE REVIEW

**Comment by mvader** [comment]
Also, seems that 
```
=== RUN   TestRateRelative
	/metric_test.go:113: TestRateRelative -> Expected int64(3), got int64(2)
```

Is broken in Go 1.6.

File: tcontainer/arrays.go [hunk 1]
```diff
@@ -62,9 +60,7 @@ func (s Uint64Slice) Less(i, j int) bool {
 }
 
 func (s Uint64Slice) Swap(i, j int) {
-	s[i] = s[i] ^ s[j]
-	s[j] = s[j] ^ s[i]
-	s[i] = s[i] ^ s[j]
+	s[i], s[j] = s[j], s[i]
 }
```
Comment (arnecls) [inline tcontainer/arrays.go]: I agree your way is more readable (so I generally prefer it), but the xor version _should_ be faster.
I never measured it though ... will do, then merge.

Comment (arnecls) [inline tcontainer/arrays.go]: There is a lot more assembly generated by the compiler for the xor version so although I could not measure any difference more instructions = not good.

**Comment by arnecls** [comment]
Thanks for participating 👍

**Pull request merged**